staynomad/Nomad-Back
# Split signup and login with Google functionality

# Relevant issue

Closes #145

# Summary of change


 - Separated **signup** and **login** with Google functionality
 - Added validation for signing up with Google prior to logging in (no longer automatically signs a user up on login page)
 - Moved user housekeeping increment and verification email functionality to helper files

# Testing/Verification

### Test in conjunction with frontend:
**1. Verify that new users cannot create an account on login page**
   - Select "Log in with Google" on the login page. 
   - Sign in with a new account (cannot be an account already registered in the database). 
   - VERIFY: the following error should display
![image](https://user-images.githubusercontent.com/66970460/124203692-240f6080-da92-11eb-98dc-67114fe3219e.png)

**2. Verify that signing up with Google is compatible with "Are you a host?" button**
   - Check the "Are you a host?" button on the sign up page.
   - Select "Sign up with Google" and sign in with a new account.
   - Once signed in, navigate to MyAccount.
   - VERIFY: MyAccount should include a tab called "MyListings"
   - VERIFY: the email associated with the account should receive the following email
![image](https://user-images.githubusercontent.com/66970460/124203967-c4658500-da92-11eb-9bd6-9c1af32f7900.png)

**3. Verify that signing up with Google increments the user housekeeping data**
 - After step 2, check the MongoDB "housekeeping" collection.
 - VERIFY: the payload for the current date in the users document should have incremented
![image](https://user-images.githubusercontent.com/66970460/124204210-6dac7b00-da93-11eb-8a19-a83e29027c8d.png)
